src/frontend/src/binder/insert.rs

```diff
@@ -102,14 +102,9 @@ impl Binder {
         let table_version_id = table_catalog.version_id().expect("table must be versioned");
         let columns_to_insert = table_catalog.columns_to_insert().cloned().collect_vec();
 
-        let expected_types: Vec<DataType> = columns_to_insert
```

Comment:
the original expected_types is derived from the target table's column layout, which is a bug.
in queries like this 
```sql
create table t (v1 real, v2 int, v3 varchar);
insert into t (v2, v1, v3) values (1, 2, 'a'), (3, 4, 'b');
```
the expected types should follow the target column identifiers 

>  (v2, v1, v3)